src/modules/tokens/hooks/useEnoughBalance.ts

```diff
@@ -0,0 +1,66 @@
+import { Currency, CurrencyAmount, Token } from '@uniswap/sdk-core'
+
+import useNativeCurrency from 'lib/hooks/useNativeCurrency'
+import { getAddress } from 'utils/getAddress'
+import { isEnoughAmount } from 'utils/isEnoughAmount'
+
+import { useBalancesAndAllowances } from './useBalancesAndAllowances'
+import { useCurrencyBalances } from './useCurrencyBalance'
+
+import { TokenAmounts } from '../types'
+
+export interface UseEnoughBalanceParams {
+  account?: string
+  amount?: CurrencyAmount<Currency>
+  checkAllowanceAddress?: string
+}
+
+export function useEnoughBalance(params: UseEnoughBalanceParams): boolean {
+  const { account, amount, checkAllowanceAddress } = params
+  const isNativeCurrency = amount?.currency.isNative
+  const token = amount?.currency.wrapped
+
+  const { balances, allowances } = useBalancesAndAllowances({
+    account,
+    spender: checkAllowanceAddress,
+    tokens: !isNativeCurrency && account && token ? [token as Token] : [],
+  })
+
+  const native = useNativeCurrency()
+  const [nativeBalance] = useCurrencyBalances(isNativeCurrency ? account : undefined, [native])
+
+  return hasEnoughBalance({
+    ...params,
+    balances,
+    allowances: checkAllowanceAddress ? allowances : undefined,
+    nativeBalance,
+  })
+}
+
+export interface EnoughBalanceParams extends Omit<UseEnoughBalanceParams, 'checkAllowanceAddress'> {
+  balances: TokenAmounts
+  allowances?: TokenAmounts
+  nativeBalance?: CurrencyAmount<Currency>
+}
+
+export function hasEnoughBalance(params: EnoughBalanceParams): boolean {
+  const { account, amount, balances, nativeBalance, allowances } = params
+  const isNativeCurrency = amount?.currency.isNative
+  const token = amount?.currency.wrapped
+  const tokenAddress = getAddress(token)
+
+  const balance = tokenAddress && balances[tokenAddress]?.value
+  const allowance = tokenAddress && allowances && allowances[tokenAddress]?.value
+
+  if (!account || !amount) {
+    return false
+  }
+
+  const balanceAmount = isNativeCurrency
+    ? nativeBalance
+    : (balance && typeof balance !== 'string' && balance) || undefined
+  const enoughBalance = isEnoughAmount(amount, balanceAmount)
+  const enoughAllowance = (allowance && typeof allowance !== 'string' && isEnoughAmount(amount, allowance)) || false
+
+  return enoughBalance && (allowances === undefined || enoughAllowance)
```

Comment:
The function checks balances AND allowances, I think it should be displayed in the function name

Reply:
Yep, good point. 
I just didn't want to overcomplicate the name, since the allowance is optional and we can consider the allowance+balance as an "effective balance". I will rename to be more explicit

Reply:
I callled it `useEnoughBalanceAndAllowance` now, but what i don't like of this name is that it makes you think it necessarily checks the allowance (when is opt-in)

Anyways, if someone has a better name, let me know

Reply:
your wish is my command 395defc080c75450dc6a5ba2315a0d6f18d8fa53

<img width="902" alt="image" src="https://github.com/cowprotocol/cowswap/assets/2352112/8c957ca1-9767-401d-b502-b31ae510dad0">